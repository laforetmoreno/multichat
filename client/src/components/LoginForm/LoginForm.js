import React from 'react';
import { func, string } from 'prop-types';

import Input from '../Input';
import Button from '../Button';
import Form from '../Form';

import styles from './UserForm.module.scss';

const LoginForm = ({ onSubmit, title = 'Chat IO', btnLabel = 'Login', ...props }) => (
  <Form onSubmit={onSubmit} className={styles.wrapper}>
    <h3 className={styles.title}>{title}</h3>
    <h5>Crie um usu√°rio para ter acesso as salas</h5>
    <Input placeholder="Digite seu apelido" {...props} />
    <Button onClick={onSubmit} className={styles.btn}>
      {btnLabel}
    </Button>
  </Form>
);

LoginForm.propTypes = {
  onSubmit: func,
  title: string,
  btnLabel: string,
};

export default LoginForm;
