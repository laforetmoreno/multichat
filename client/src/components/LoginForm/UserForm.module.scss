@import '../../styles/variables';

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 300px;

  @media (max-width: 768px) {
    width: 100%;
  }

  .title {
    font-size: 26px;
    text-align: center;
  }

  .btn {
    margin: 20px 0;
    width: 100%;
  }
}
