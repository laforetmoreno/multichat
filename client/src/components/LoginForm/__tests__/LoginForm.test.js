import React from 'react';
import LoginForm from '../LoginForm';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('<LoginForm />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<LoginForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('testing events', async () => {
    const { getByTestId } = render(<LoginForm />)
      const fieldInput = await waitForElement(() => getByTestId('input'))
      const value = 'test value'
      fireEvent.change(fieldInput, { target: { value } })

      const fieldBtn = await waitForElement(() => getByTestId('btn'))
      fireEvent.submit(fieldBtn)

      expect(fieldInput.value).toEqual(value)
  });
});