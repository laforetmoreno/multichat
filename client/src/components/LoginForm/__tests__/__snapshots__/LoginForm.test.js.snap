// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders correctly 1`] = `
<form
  className="wrapper"
>
  <h3
    className="title"
  >
    Chat IO
  </h3>
  <h5>
    Crie um usu√°rio para ter acesso as salas
  </h5>
  <input
    className="wrapper "
    data-testid="input"
    placeholder="Digite seu apelido"
  />
  <button
    className="btn primary"
    data-testid="btn"
  >
    Login
  </button>
</form>
`;
